jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -2416,6 +2416,14 @@ ServletPathMapping findServletPathMapping()
     @Override
     public HttpServletMapping getHttpServletMapping()
     {
+        // TODO This is to pass the current TCK.  This has been challenged in https://github.com/eclipse-ee4j/jakartaee-tck/issues/585
+        if (_dispatcherType == DispatcherType.ASYNC)
+        {
+            Object async = getAttribute(AsyncContext.ASYNC_MAPPING);
+            if (async != null)
```

Comment:
Do we really want to fall through to use the target mapping if this is null.
Should we skip the null check here and return `(ServletPathMapping)async` always.

Reply:
I think it is a mute point as the async mapping should never be null when the dispatch type is ASYNC... however, just in case  I would prefer to return non null.